hcipy/aperture/generic.py

```diff
@@ -362,6 +376,23 @@ def make_rotated_aperture(aperture, angle):
 	'''
 	return lambda grid: Field(aperture(grid.rotated(-angle)), grid)
 
+def make_shifted_aperture(aperture, shift):
+	'''Create a shifted version of `aperture`.
+
+	Parameters
+	----------
+	aperture : Field generator
+		The aperture to rotate.
+	shift : array_like
+		The shift of the aperture.
+
+	Returns
+	-------
+	Field generator
+		The shifted aperture.
+	'''
+	return lambda grid: Field(aperture(grid.shifted(-np.array(shift)), grid))
```

Comment:
This is the only test failure. This should probably be:
```python
return lambda grid: Field(aperture(grid.shifted(-np.array(shift))))
```

Reply:
yeah, I see the error now. but you need to include the original grid for the field. Because its not the grid that has been shifted, its the aperture.